File: menu.py

class Menu:
    @staticmethod
    def logIn(store):
        print("|| BIENVENIDO AL SISTEMA DE LA TIENDA! ||")
        print("\nPorfavor ingrese la contrasena para acceder al sistema: ")
        password = input()

        while password != "tienda123":
            print("\n|| Contrasena incorrecta, intente de nuevo: ||")
            password = input()

        print("\n|| Ha ingresado la contrasena correcta!! ||")
        Menu.firstMenu(store)

    @staticmethod
    def firstMenu(store):
        flag = True

        while flag:
            print("\n== QUE DESEA HACER EL DIA DE HOY ==")
            print("1. Registrar")
            print("2. Imprimir datos")
            print("3. Modificar productos")
            print("4. Eliminar")
            print("5. Realizar compra")
            print("6. Cerrar programa")

            opt = int(input())

            if opt == 1:
                Menu.register(store)
            elif opt == 2:
                Menu.prints(store)
            elif opt == 3:
                Menu.modify(store)
            elif opt == 4:
                Menu.delete(store)
            elif opt == 5:
                Menu.buyOrder(store)
            elif opt == 6:
                flag = False
                print("\n== GRACIAS POR USAR EL SISTEMA ==")
            else:
                print("Opcion ingresada incorrecta!")

    @staticmethod
    def register(store):
        flag = True

        while flag:
            print("\n== QUE DESEA REGISTRAR: ==")
            print("1. Producto")
            print("2. Cliente")
            print("3. Volver")

            opt = int(input())

            if opt == 1:
                store.registerProduct()
            elif opt == 2:
                store.registerClient()
            elif opt == 3:
                flag = False
            else:
                print("Opcion ingresada incorrecta!")

    @staticmethod
    def prints(store):
        flag = True

        while flag:
            print("\n== QUE DESEA IMPRIMIR: ==")
            print("1. Productos maquillaje")
            print("2. Productos comida")
            print("3. Productos limpieza")
            print("4. Productos electrodomesticos")
            print("5. Clientes")
            print("6. Compras")
            print("7. Volver al menu")

            opt = int(input())

            if opt == 1:
                store.printMakeup()
            elif opt == 2:
                store.printFood()
            elif opt == 3:
                store.printCleaning()
            elif opt == 4:
                store.printElectrodomestico()
            elif opt == 5:
                store.printClients()
            elif opt == 6:
                store.printOrder()
            elif opt == 7:
                flag = False
            else:
                print("Opcion ingresada incorrecta!")

    @staticmethod
    def modify(store):
        flag = True

        while flag:
            print("\n== QUE DESEA MODIFICAR ==")
            print("1. Productos")
            print("2. Clientes")
            print("3. Volver al menu")

            opt = int(input())

            if opt == 1:
                store.modifyProduct()
            elif opt == 2:
                store.modifyClient()
            elif opt == 3:
                flag = False
            else:
                print("Opcion ingresada incorrecta!")

    @staticmethod
    def delete(store):
        flag = True

        while flag:
            print("\n== QUE DESEA ELIMINAR ==")
            print("1. Productos")
            print("2. Clientes")
            print("3. Volver al menu")

            opt = int(input())

            if opt == 1:
                store.deleteProduct()
            elif opt == 2:
                store.deleteClient()
            elif opt == 3:
                flag = False
            else:
                print("Opcion ingresada incorrecta!")

    @staticmethod
    def buyOrder(store):
        print("\n== INGRESE LOS DATOS DE LA COMPRA ==")
        store.buyOrder()

File: Sys/Menu.java
package Sys;
import java.util.Scanner;
public class Menu {
    private static Scanner sc = new Scanner(System.in);
    public static void logIn(Store store){
        System.out.println("|| BIENVENIDO AL SISTEMA DE LA TIENDA! ||");
        System.out.println("\nPorfavor ingrese la contrasena para acceder al sistema: ");
        String pass = sc.next();

        while (!pass.equals("tienda123")){
            System.out.println("\n|| Contrasena incorrecta, intente de nuevo: ||");
            pass = sc.next();
        }

        System.out.println("\n|| Ha ingresado la contrasena correcta!! ||");
        firstMenu(store);
    }

    private static void firstMenu(Store store){
        boolean flag = true;

        while (flag) {
            System.out.println("\n== QUE DESEA HACER EL DIA DE HOY ==");
            System.out.println("1. Registrar");
            System.out.println("2. Imprimir datos");
            System.out.println("3. Modificar productos");
            System.out.println("4. Eliminar");
            System.out.println("5. Realizar compra");
            System.out.println("6. Cerrar programa");

            int opt = sc.nextInt();

            switch (opt) {
                case 1 -> register(store);
                case 2 -> prints(store);
                case 3 -> modify(store);
                case 4 -> delete(store);
                case 5 -> buyOrder(store);
                case 6 -> {
                    flag = false;
                    System.out.println("\n== GRACIAS POR USAR EL SISTEMA ==");
                }
                default -> System.out.println("Opcion ingresada incorrecta!");
            }

        }
    }

    private static void register(Store store){
        boolean flag = true;

        while (flag) {
            System.out.println("\n== QUE DESEA REGISTRAR: ==");
            System.out.println("1. Producto");
            System.out.println("2. Cliente");
            System.out.println("3. Volver");

            int opt = sc.nextInt();

            switch (opt) {
                case 1 -> store.registerProduct();
                case 2 -> store.registerClient();
                case 3 -> flag = false;
                default -> System.out.println("Opcion ingresada incorrecta!");
            }
        }
    }

    private static void prints(Store store){
        boolean flag = true;

        while (flag) {
            System.out.println("\n== QUE DESEA IMPRIMIR: ==");
            System.out.println("1. Productos maquillaje");
            System.out.println("2. Productos comida");
            System.out.println("3. Productos limpieza");
            System.out.println("4. Productos electrodomesticos");
            System.out.println("5. Clientes");
            System.out.println("6. Compras");
            System.out.println("7. Volver al menu");

            int opt = sc.nextInt();

            switch (opt) {
                case 1 -> store.printMakeup();
                case 2 -> store.printFood();
                case 3 -> store.printCleaning();
                case 4 -> store.printElectrodomestico();
                case 5 -> store.printClients();
                case 6 -> store.printOrder();
                case 7 -> flag = false;
                default -> System.out.println("Opcion ingresada incorrecta!");
            }
        }
    }

    private static void modify(Store store){
        boolean flag = true;
        while (flag) {
            System.out.println("\n== QUE DESEA MODIFICAR ==");
            System.out.println("1. Productos");
            System.out.println("2. Clientes");
            System.out.println("3. Volver al menu");

            int opt = sc.nextInt();

            switch (opt) {
                case 1 -> store.modifyProduct();
                case 2 -> store.modifyClient();
                case 3 -> flag = false;
                default -> System.out.println("Opcion ingresada incorrecta!");
            }
        }
    }

    private static void delete(Store store){
        boolean flag = true;
        while (flag) {
            System.out.println("\n== QUE DESEA ELIMINAR ==");
            System.out.println("1. Productos");
            System.out.println("2. Clientes");
            System.out.println("3. Volver al menu");

            int opt = sc.nextInt();

            switch (opt) {
                case 1 -> store.deleteProduct();
                case 2 -> store.deleteClient();
                case 3 -> flag = false;
                default -> System.out.println("Opcion ingresada incorrecta!");
            }
        }
    }

    private static void buyOrder(Store store){
        System.out.println("\n== INGRESE LOS DATOS DE LA COMPRA ==");
        store.buyOrder();
    }
}
